### DISCUSSION TITLE
"virtual inverter" request

### DISCUSSION THREAD

**Discussion by the-butterfry** [WillCodeForCats/solaredge-modbus-multi#208] [Ideas]
@WillCodeForCats we spoke over on the HA forums about multi-inverter setups and creating a virtual inverter for use as an overall reading of the entire site. 

I've been mucking around with Remko's SE Dashboard setup, and the configuration for more than a few inverters gets daunting and messy pretty quick. Also, I've been struggling creating usable templates that mimic a combined virtual inverter. In theory, I should be able to create template sensors from the aggregated i1/i2/i3/etc.. data streams. It doesn't seem to be working for me. Certain I'm just missing something.. but then I thought this will become a configuration headache the more systems/variations that I attempt to setup. I think it might just be easier if you did the work (heh). In this case I (and even single inverter setups) can just have one config that works for everything. Instead of a single setup using i1 in the config, you could just use that virtual inverter that would represent everything on the modbus chain. 

Here's the request: An aggregated 'virtual universal' inverter with a friendly name. Instead of i1/2/3 it would be solaredge_site_XXXXXX named sensors. The resulting inverter could just be used in lieu of adding up to 32 individual sensor lines, or creating big template sensors (that I can't seem to figure out) to feed into the dashboard. You could also have a Configuration switch in the setup that enables this virtual inverter, so those that only have one can still use i1/b1/m1 and don't get stats/sensor pollution.

I'm a SolarEdge commercial and resi installer - so I have access to a number of large and small sites with and without battery backup for testing. 

BONUS: **[EDIT: I think this HA install is borked - I've been messing with it so much. My changes in yaml aren't showing up in the entity editor. Starting over... ]**

Here's the issue I'm having - I can create a template sensor just fine, that reads in Dev Tools fine - but as soon as I swap out the code of the single i1_dc_power sensor, it won't work.

My template sensor:

```
      - name: "Site Solar DC Power"
        unique_id: solaredge_site_dc_power
        state_class: measurement
        device_class: power
        unit_of_measurement: "W"
        icon: mdi:solar-power
        state: >    
          {{ [ states('sensor.solaredge_i1_dc_power'), 
               states('sensor.solaredge_i2_dc_power'),
               states('sensor.solaredge_i3_dc_power'), 
               states('sensor.solaredge_i4_dc_power'),
               states('sensor.solaredge_i5_dc_power'),
               states('sensor.solaredge_i6_dc_power'), 
               states('sensor.solaredge_i7_dc_power') ]
               | map('float') | sum }}
        availability: >
          {{ not 'unavailable' in
             [ states('sensor.solaredge_i1_dc_power'), 
               states('sensor.solaredge_i2_dc_power'),
               states('sensor.solaredge_i3_dc_power'), 
               states('sensor.solaredge_i4_dc_power'),
               states('sensor.solaredge_i5_dc_power'),
               states('sensor.solaredge_i6_dc_power'), 
               states('sensor.solaredge_i7_dc_power') ] }}   
```

result:
```
state: >    
          6677.9
        availability: >
          True
```

BUT when I plug in this to Remko's yaml for solar_panel_production_w i get nada. 

```
    - name: "Solar Panel Production W"
      unique_id: solar_panel_production_w
      unit_of_measurement: "W"
      icon: mdi:solar-power  
      state: >
          {% set i1_dc_power = states('sensor.solaredge_site_dc_power') | float(0) %}

          {% if (is_state('sensor.solaredge_site_dc_power', 'unknown')) %}
            0
          {% elif (i1_dc_power <= 0) %}
            0
          {% else %}
            {{ (i1_dc_power) }}
          {% endif %}
        availability: >
          {{ states('sensor.solaredge_site_dc_power') | is_number and states('sensor.solaredge_site_ac_power') }} 
```
```
state: >
          

          
            0
          
        availability: >
          False
```

**Comment by the-butterfry** [comment]
K, ignore my madness about stuff not working. I started over with a fresh install and magically, it's working. I think i'd gotten HA into a funky state somehow. 

I think my aggregated template sensors are working fine now - We'll see when I get some sun.

**Comment by WillCodeForCats** [comment]
I've been looking at this recently since it was requested, and I'm not thinking this makes sense as an integration feature anymore. Home Assistant now has entity helpers that can do this plus more as a core function without having to mess with templates or states.

Example in issue #254
